client/task-list/test/index.js

```diff
@@ -0,0 +1,72 @@
+/**
+ * External dependencies
+ */
+import { render, screen } from '@testing-library/react';
+import '@testing-library/jest-dom/extend-expect';
+import apiFetch from '@wordpress/api-fetch';
+
+/**
+ * Internal dependencies
+ */
+import TaskDashboard from '../index.js';
+
+jest.mock( '@wordpress/api-fetch' );
+
+// Mock the tasks.
+jest.mock( '../tasks', () => {
+	return {
+		getAllTasks: jest.fn().mockImplementation( () => [
+			{
+				key: 'optional',
+				title: 'This task is optional',
+				container: null,
+				completed: false,
+				visible: true,
+				time: '1 minute',
+				isDismissable: true,
+			},
+			{
+				key: 'required',
+				title: 'This task is required',
+				container: null,
+				completed: false,
+				visible: true,
+				time: '1 minute',
+				isDismissable: false,
+			},
+			{
+				key: 'completed',
+				title: 'This task is completed',
+				container: null,
+				completed: true,
+				visible: true,
+				time: '1 minute',
+				isDismissable: true,
+			},
+		] ),
+	};
+} );
+
+describe( 'TaskDashboard', () => {
+	afterEach( () => jest.clearAllMocks() );
+
+	it( 'renders a dismiss button for tasks that are optional and incomplete', async () => {
+		apiFetch.mockResolvedValue( {} );
+
+		render( <TaskDashboard query={ {} } /> );
+
+		// Wait for the task list to render.
+		expect( await screen.findByText( 'Finish setup' ) ).toBeDefined();
```

Comment:
Not requesting a change but just an FYI that you mostly never need to query the whole screen for component tests. Rather you can capture the container when calling render and then query that.

e.g.

```
const {container} = render(<TaskDashboard query={{}} />);

expect( container.findByText('Finish setup')).toBeDefined();
```

Reply:
Thanks! I'll make this change.

Reply:
I am querying the container now in 985f8d3e39b9012db51fd65de74a05db76878f5b 😄